### PR TITLE
Compressed Auth

### PR DESCRIPTION
**Pull request by btkostner**
Condensed all the role checking functions into a new super function called `hasRole`. It can accept a string, or an number referring to the index of user rights.

Also added optional url redirection for when you log in, and implemented it on all existing routes. No longer are you always redirected to the dashboard!

### CODE REVIEW

File: app/controller/auth.js [hunk 1]
```diff
@@ -36,71 +36,80 @@ app.use(passport.session());
...
 // Redirect to GitHub to login
 app.get('/auth/github',
-  passport.authenticate('github', { scope: 'repo read:org' }),
-    function(req, res) {
-      // Just a redirect, no actions here.
-    });
+  passport.authenticate('github', {
+    scope: 'repo read:org',
+  }), (req, res, next) => {
+    return next(); // Never gets called
```
Comment (aroman) [inline app/controller/auth.js]: why doesn't it get called?

File: app/controller/auth.js [hunk 2]
```diff
@@ -36,71 +36,80 @@ app.use(passport.session());
...
 
 // GitHub Login callback
 // TODO: Add failure redirect
 app.get('/auth/github/callback',
   passport.authenticate('github'),
-    function(req, res) {
-      res.redirect('/dashboard');
-    });
+  (req, res, next) => {
+    let path = req.session.authredirect || '/dashboard';
```
Comment (aroman) [inline app/controller/auth.js]: const

**Comment by aroman** [comment]
Looks good! Just change the `let`s to `const`s and :ship: away :)

File: app/controller/auth.js (hunk 1 above)
Comment (btkostner) [inline app/controller/auth.js]: Because it always gets redirected to GitHub for authentication, which then gets redirected to the callback URL.

**Pull request merged**